libs/domain/picking/src/components/picking-lists/picking-lists.component.ts

```diff
@@ -1,24 +1,38 @@
 import { resolve } from '@spryker-oryx/di';
 import { asyncValue } from '@spryker-oryx/utilities';
 import { html, LitElement, TemplateResult } from 'lit';
+import { state } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
 import { repeat } from 'lit/directives/repeat.js';
 import { when } from 'lit/directives/when.js';
 import { PickingList, PickingListStatus } from '../../models';
 import { PickingListService } from '../../services';
+import { styles } from './picking-lists.styles';
 
 export class PickingListsComponent extends LitElement {
+  static styles = styles;
+
+  @state() customerNote?: string;
+
   protected pickingListService = resolve(PickingListService);
 
   protected pickingLists$ = this.pickingListService.get({
     status: PickingListStatus.ReadyForPicking,
   });
 
-  protected override render(): unknown {
-    return html`${asyncValue(
-      this.pickingLists$,
-      (pickingLists) => this.renderPickingLists(pickingLists),
-      () => html`<p>Loading picking lists...</p>`
-    )}`;
+  protected override render(): TemplateResult {
+    return html`
+      ${asyncValue(
+        this.pickingLists$,
+        (pickingLists) => this.renderPickingLists(pickingLists),
+        () => html`<p>Loading picking lists...</p>`
+      )}
+
+      <oryx-customer-note-modal
```

Comment:
And I'm wondering, why do we need this modal as separate component? It is not so complex inside, and to decrease complexity of the list will be better just use oryx-modal here

Reply:
The customer note modal looks like a separate component and has different responsibilities than picking lists component. Those are reasons why it should be a separate component.

Reply:
This should be true in case if:
1) Component will be reused inside same template or even in another components
2) To decrease complexity of parent component and make it much more readable

Seems to be both cases weren't matched. I won't be reused and complexity even grew.

In result customer note modal contains a few rows of template, useless event dispatcher and doesn't contain any other specific stuff. Even it's styles should be aligned on app level (excepts paddings of the card). I would strongly suggest you just move it's template part to the picking card directly


```
protected override render(): TemplateResult {
    return html`
      ${this.renderPickingLists()}

       <oryx-modal 
         ?open=${this.customerNote}
          enableFooter
          @oryx.close=${this.closeCustomerNoteModal}
       >
          <span slot="heading">${i18n('picking.customer-note.heading')}</span/>
          ${this.customerNote}
          <oryx-button slot="footer" type="primary" size="small">
            <button value="cancel">
              <oryx-icon type="checkMark"></oryx-icon>
              ${i18n('picking.customer-note.close')}
            </button>
          </oryx-button>
        </oryx-modal>
    `;
  }
```